python/ray/experimental/workflow/tests/test_object_deref.py

```diff
@@ -0,0 +1,82 @@
+from typing import List, Dict
+
+import pytest
+
+import ray
+from ray.experimental import workflow
+
+# alias because the original type is too long
+RRef = ray.ObjectRef
+
+
+@ray.remote
+def nested_ref():
+    return ray.put(42)
+
+
+@workflow.step
+def nested_workflow(n: int):
+    if n <= 0:
+        return "nested"
+    else:
+        return nested_workflow.step(n - 1)
+
+
+@workflow.step
+def deref_check(u: int, v: "RRef[int]", w: "List[RRef[RRef[int]]]", x: str,
+                y: List[str], z: List[Dict[str, str]]):
+    try:
+        return (u == 42 and ray.get(v) == 42 and ray.get(ray.get(w[0])) == 42
+                and x == "nested" and y[0] == "nested"
+                and z[0]["output"] == "nested")
+    except Exception:
+        return False
+
+
+@workflow.step
+def deref_shared(x, y):
+    # x and y should share the same variable.
+    x.append(2)
+    return y == [1, 2]
+
+
+@workflow.step
+def empty_list():
+    return [1]
+
+
+@ray.remote
+def receive_workflow(workflow):
+    pass
+
+
+@ray.remote
+def return_workflow():
+    return empty_list.step()
+
+
+def test_object_deref():
```

Comment:
Do we have the test to cover this case: in the workflow, it returns an object ref. For example, like this one:

```
@workflow.step
def return_data() -> ObjectRef:
    obj = ray.put(create_big_data())
    return obj # OK

@workflow.step
def receive_data(data: Array):
    print(data)

@workflow.step
def entry():
    obj = return_data.step()
    receive_data.step(obj)
```

Reply:
I add the test